File: src/main/java/com/example/desafio/api/models/entitys/CarEntity.java
package com.example.desafio.api.models.entitys;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "cars")
public class CarEntity extends BaseEntity {
//    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
//    private Long id;

    @Column(name = "timestamp_register")
    private LocalDateTime timestampRegister;

    private Integer year;

    private String fuel;

    @Column(name = "num_doors")
    private Integer numDoors;

    private String color;

    @ManyToOne
    @JoinColumn(name = "model_id")
    private ModelCarEntity modelId;
}

File: src/main/kotlin/com/example/desafio/api/models/entitys/CarEntity.kt
package com.example.desafio.api.models.entitys

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
@Table(name = "cars")
data class CarEntity(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        var deleted: Boolean = false,

        @Column(name = "timestamp_register")
        var timestampRegister: Long? = null,

        var year: Int? = null,

        var fuel: String? = null,

        @Column(name = "num_doors")
        var numDoors: Int? = null,

        var color: String? = null,

        @JsonIgnore
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "model_id")
        var modelId: ModelCarEntity? = null
)
